File: 2020/17/Haskell/src/Main.hs
import qualified Part1                         as P1
import qualified Part2                         as P2
import           System.Environment
import           Text.Printf

output :: String -> IO ()
output path = do
  input <- readFile path
  printf "File: %s\n" path
  printf "  Part 1: %d\n" . P1.part1 $ input
  printf "  Part 2: %d\n" . P2.part2 $ input

main :: IO ()
main = getArgs >>= mapM_ output

File: 2020/17/Rust/src/main.rs
mod part_1;
mod part_2;

use crate::part_1 as P1;
use crate::part_2 as P2;

use std::env;
use std::fs;

fn output(path: &str) {
    let input = fs::read_to_string(&path).unwrap();
    let part_1 = P1::init_grid(&input);
    let part_2 = P2::init_grid(&input);

    println!("File: {}", path);
    println!("  Part 1: {}", P1::cycle_grid(&part_1));
    println!("  Part 2: {}", P2::cycle_grid(&part_2));
}

fn main() {
    env::args()
        .skip(1)
        .collect::<Vec<_>>()
        .iter()
        .for_each(|x| output(x));
}
